lib/index.js

```diff
@@ -147,6 +147,7 @@ Mongoose.prototype.driver = require('./driver');
  *
  * Currently supported options are:
  * - 'debug': If `true`, prints the operations mongoose sends to MongoDB to the console. If a writable stream is passed, it will log to that stream, without colorization. If a callback function is passed, it will receive the collection name, the method name, then all arugments passed to the method. For example, if you wanted to replicate the default logging, you could output from the callback `Mongoose: ${collectionName}.${methodName}(${methodArgs.join(', ')})`.
+ * - 'defaultNewOnFindAndUpdateOrReplace': If `true`, changes the default `new` option to `findOneAndUpdate()`, `findByIdAndUpdate` and `findOneAndReplace()` to true.
```

Comment:
I like this change, but I think the name is a bit too verbose and isn't consistent with other naming conventions: we typically give global, schema-level, and connection-level options the same name as their child options.

Admittedly, `mongoose.set('new', true)` suffers from the opposite problem: `new` is too generic to be a global option name.

Maybe we instead name the global option `returnOriginal`? We already support a `returnOriginal` option, so perhaps to opt in to always getting the equivalent of `new: true` we ask users to do `mongoose.set('returnOriginal', false)`?

Reply:
I agree, I started with `defaultNewOnFindAndUpdate` but I was concerned that users would expect it to not work on `findOneAndReplace`, so I added the last bit, which became too verbose. I justified that by the fact that this is written only once in a project, so I leaned toward the more verbose name.

However, `returnOriginal` has an advantage, it is clear enough to deliver the meaning and generic enough that we can apply the option to more operations in the future, add the changes in the documentation, and users can then look at it later. We're not limiting ourselves by the option name.